ui/hooks/useTransactionDisplayData.js

```diff
@@ -217,17 +217,20 @@ export function useTransactionDisplayData(transactionGroup) {
     title = t('approveSpendLimit', [token?.symbol || t('token')]);
     subtitle = origin;
     subtitleContainsOrigin = true;
-  } else if (
-    type === TRANSACTION_TYPES.DEPLOY_CONTRACT ||
-    type === TRANSACTION_TYPES.CONTRACT_INTERACTION
-  ) {
+  } else if (type === TRANSACTION_TYPES.CONTRACT_INTERACTION) {
     category = TRANSACTION_GROUP_CATEGORIES.INTERACTION;
     const transactionTypeTitle = getTransactionTypeTitle(t, type);
     title =
       (methodData?.name && camelCaseToCapitalize(methodData.name)) ||
       transactionTypeTitle;
```

Comment:
Do we ever add `methodData` to a transaction where the `to` address is `null`/`undefined`?

Reply:
We shouldn't. We should only use it for valid method calls to a contract.

Reply:
I feel like ideally we would guarantee that methodData won't be attached to a `transaction` where address is empty or zero address further upstream (maybe in the `TransactionsController`), but won't block on it here...

Reply:
I feel like ideally we would guarantee that methodData won't be attached to a transaction where address is empty or zero address further upstream (maybe in the TransactionsController), but won't block on it here...

Reply:
in what scenario would we have a transaction without a 'to' address?

Reply:
I think when you deploy a contract no?

Reply:
🤦🏻 yeah. that makes sense...

Reply:
We don't attach methoData anywhere, its fetched at that top of the useTransactionDisplayData hook.

Reply:
Yeah, it's not affiliated with the transaction outside of this hook afaik @adonesky1